### PR TITLE
Sui Native bridge

### PR DESCRIPTION
**Pull request by patrickkuo**
## Description 

Sui Native Bridge v1 for bridging asset from Ethereum to Sui

## Test plan 

How did you test the new or updated feature?

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [x] Protocol: Added Sui native bridge package `0xb` to the Sui framework, the bridge object `0x9` will be created in the next epoch after protocol upgrade.
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [ ] CLI: 
- [ ] Rust SDK:

### CODE REVIEW

**Review by dariorussi — APPROVED** [review]
this looks good to me, though I have not checked and cannot check all code.
It is essentially the bridge code as coming from the branch so not too clear what we need to review.
If anybody wanted to take a look please go ahead, we are trying to get it in by today.
A cursory look for the parts you may care about is good too.
Thanks

File: crates/sui-protocol-config/src/lib.rs [hunk 1]
```diff
@@ -13,7 +13,7 @@ use tracing::{info, warn};
 
 /// The minimum and maximum protocol versions supported by this build.
 const MIN_PROTOCOL_VERSION: u64 = 1;
-const MAX_PROTOCOL_VERSION: u64 = 45;
+const MAX_PROTOCOL_VERSION: u64 = 46;
```
Comment (patrickkuo) [inline crates/sui-protocol-config/src/lib.rs:16]: @ebmifa do I need to increment the protocol version to 46 for next devnet cut?

Comment (ebmifa) [reply crates/sui-protocol-config/src/lib.rs:16]: We are already at `45` in main, why are we skipping to `46`? 
CC: @amnn

Comment (patrickkuo) [reply crates/sui-protocol-config/src/lib.rs:16]: I will revert it back to 45, I didn't know if 45 is already deployed or not.

Comment (ebmifa) [reply crates/sui-protocol-config/src/lib.rs:16]: Yeah, `45` is already in `devnet` as of this Monday.

**Pull request merged**